sklearn/base.py

```diff
@@ -311,16 +311,36 @@ def __setstate__(self, state):
             self.__dict__.update(state)
 
     def _get_tags(self):
+        if '_more_tags' in self.__class__.__dict__:
+            # we cannot use getattr since it would resolve to the parent's
+            # attribute if self.__class__ does not overrides it. So we use
+            # __dict__ instead.
+            overridden_tags = self._more_tags()
+        else:
+            overridden_tags = {}
+
+        def _resolve_tags(c):
+            nonlocal collected_tags
```

Comment:
can't you just pass `collected_tags` as a reference to `_resolve_tags`?

Reply:
Sure